lib/systems/default.nix

```diff
@@ -2,4 +2,22 @@ rec {
   doubles = import ./doubles.nix;
   parse = import ./parse.nix;
   platforms = import ./platforms.nix;
+
+  # Elaborate a `localSystem` or `crossSystem` so that it contains everything
+  # necessary.
+  #
+  # `parsed` is inferred from args, both because there are two options with one
+  # clearly prefered, and to prevent cycles. A simpler fixed point where the RHS
+  # always just used `final.*` would fail on both counts.
+  elaborate = args: let
+    final = {
+      # Prefer to parse `config` as it is strictly more informative.
+      parsed = parse.mkSystemFromString (if args ? config then args.config else args.system);
```

Comment:
What is this `config` attribute and from where does it come from?

Reply:
Sadly that wiki page is gone now - Can we not rename this attribute to `llvmTargetConfig`?

Reply:
https://web.archive.org/web/20160322211219/https://nixos.org/wiki/CrossCompiling

I'm all for improving the name but we must be careful as this is a public interface to how Nixpkgs is instantiated.

Also, while I follow LLVM most closely, other tools use these too.

Reply:
ah right. In that case, maybe rename `args` to `compileConfig`?